### PR TITLE
chore: pre-commit (black, isort, mypy, flake8)

### PR DESCRIPTION
**Pull request by arys-splunk**
I have also added myself as a CODEOWNER for `.pre-commit-config.yaml`.

### CODE REVIEW

File: .flake8 [hunk 1]
```diff
@@ -0,0 +1,3 @@
+[flake8]
+ignore = E203, E266, E501, W503, F403, F401, E402
```
Comment (arys-splunk) [inline .flake8:2]: @omrozowicz-splunk I am not sure that we need to ignore so many codes.

After running `pre-commit run flake8 --all-files`:

```
flake8...................................................................Failed
- hook id: flake8
- exit code: 1

splunk_connect_for_snmp_poller/manager/static/mib_enricher.py:27:82: E203 whitespace before ':'
tests/test_tasks.py:23:1: E402 module level import not at top of file
tests/test_tasks.py:24:1: E402 module level import not at top of file
tests/test_task_utilities.py:95:121: E501 line too long (338 > 120 characters)
```

We can ignore them inline, like:
```
example = lambda: 'example'  # noqa: E731
```

More info: https://flake8.pycqa.org/en/3.1.1/user/ignoring-errors.html

Comment (omrozowicz-splunk) [reply .flake8:2]: Good point, I'll reduce the list to only 3 codes that we need for now.

File: .mypy.ini [hunk 2]
```diff
@@ -0,0 +1,5 @@
+[mypy-requests.*]
```
[the remark this answers is not in the document]
Comment (arys-splunk) [reply .mypy.ini]: N/A anymore.

**Review by omrozowicz-splunk — APPROVED** [review]

**Pull request merged**